src/hooks/useSyncChainQuery.ts

```diff
@@ -40,14 +41,20 @@ export default function useSyncChainQuery() {
 
   const [searchParams, setSearchParams] = useSearchParams()
 
-  const chainQueryManuallyUpdated = urlChainId && urlChainId !== previousUrlChainId && isActive
+  useEffect(() => {
+    const chainQueryManuallyUpdated = urlChainId && urlChainId !== previousUrlChainId
+    if (chainQueryManuallyUpdated) {
+      setNextChainId(urlChainId)
+    }
+  }, [previousUrlChainId, urlChainId])
 
   return useEffect(() => {
-    if (chainQueryManuallyUpdated) {
+    if (nextChainId && isActive) {
       // If the query param changed, and the chain didn't change, then activate the new chain
-      selectChain(urlChainId)
+      selectChain(nextChainId)
       searchParams.delete('chain')
       setSearchParams(searchParams)
+      setNextChainId(undefined)
```

Comment:
wondering if this would work better, since the chain switch might be rejected or unsupported and we wouldn't want to update the url in that case
```suggestion
      selectChain(nextChainId).then(() => {
        searchParams.delete('chain')
        setSearchParams(searchParams)
        setNextChainId(undefined)
      })
```